File: find_first_and_last_position.py
def find_first_and_last_position(nums, target):
    def binary_search_left(nums, target):
        left, right = 0, len(nums) - 1
        while left <= right:
            mid = (left + right) // 2
            if nums[mid] < target:
                left = mid + 1
            else:
                right = mid - 1
        return left

    def binary_search_right(nums, target):
        left, right = 0, len(nums) - 1
        while left <= right:
            mid = (left + right) // 2
            if nums[mid] <= target:
                left = mid + 1
            else:
                right = mid - 1
        return right

    left_index = binary_search_left(nums, target)
    right_index = binary_search_right(nums, target)

    if left_index <= right_index:
        return [left_index, right_index]
    else:
        return [-1, -1]

#Example usage:
nums = [5, 7, 8, 9, 10]
target = 100
print(find_first_and_last_position(nums, target))  # Output: [3, 4]

target = 6
print(find_first_and_last_position(nums, target))  # Output: [-1, -1]

File: FindFirstAndLastPosition.java
class Solution {

    private int searchFirstValue(int start, int end, int[] nums, int target) {

        while(start <= end) {
            int mid = start + (end - start)/2;
            if(nums[mid] == target) {
                if(mid ==0 || nums[mid - 1] < target) {
                     return mid;
                } else {
                   end = mid -1;
                }
            } else if(nums[mid] > target) {
                end = mid - 1;
            } else {
                start = mid + 1;
            }
        }
        return -1;
    }

    private int searchSecondValue(int start, int end, int[] nums, int target) {

        while(start <= end) {
            int mid = start + (end - start)/2;
            if(nums[mid] == target) {
                if(mid == end || nums[mid + 1] > target) {
                   return mid;
                } else {
                     start = mid + 1;
                }
            } else if(nums[mid] > target) {
                end = mid - 1;
            } else {
                start = mid + 1;
            }
        }
        return -1;
    }
    public int[] searchRange(int[] nums, int target) {

        int start = 0;
        int end = nums.length - 1;

        int firstValue = searchFirstValue(start, end, nums, target);

        if(firstValue < 0) return new int[]{-1, -1};

        int secondValue = searchSecondValue(firstValue, end, nums, target);
        return new int[]{firstValue, secondValue};
        
    }
}
